packages/medusa/src/services/cart.ts

```diff
@@ -1062,7 +1062,7 @@ class CartService extends TransactionBaseService {
     const productsToKeep = await this.productService_
       .withTransaction(this.manager_)
       .filterProductsBySalesChannel(productIds, newSalesChannelId, {
-        select: ["id", "sales_channels"],
+        select: ["id"],
```

Comment:
question: Is this intentionally included in the PR?

Reply:
yes cause selecting sales_channels generate a malformed query. for now, I can't determine why but it is part of the issue I sent you

Reply:
Ah got it. Makes sense!